File: web/models/password-reset.ts
export type SendVerifyCodePayload = {
  tenant: string
  username: string
  locale: string
}

export type SendVerifyCodeResponse = {
  message: string
}

export type ResetPasswordPayload = {
  tenant: string
  username: string
  verify_code: string
  new_password: string
}

export type ResetPasswordResponse = {
  message: string
}

File: server/app/schemas/password_reset.py
"""
Password reset schemas
"""
from pydantic import BaseModel


class SendVerifyCodeRequest(BaseModel):
    tenant: str
    username: str
    locale: str = "zh"


class SendVerifyCodeResponse(BaseModel):
    message: str = "ok"


class ResetPasswordRequest(BaseModel):
    tenant: str
    username: str
    verify_code: str
    new_password: str


class ResetPasswordResponse(BaseModel):
    message: str = "ok"
